messagingpush/src/sharedTest/java/io/customer/messagingpush/CustomerIOPushNotificationHandlerTest.kt

```diff
@@ -0,0 +1,57 @@
+package io.customer.messagingpush
+
+import android.os.Bundle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.firebase.messaging.RemoteMessage
+import io.customer.commontest.BaseTest
+import io.customer.messagingpush.activity.NotificationClickReceiverActivity
+import io.customer.messagingpush.data.model.CustomerIOParsedPushPayload
+import io.customer.messagingpush.extensions.parcelable
+import io.customer.sdk.extensions.random
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.robolectric.Shadows
+
+@RunWith(AndroidJUnit4::class)
+internal class CustomerIOPushNotificationHandlerTest : BaseTest() {
+
+    private lateinit var pushNotificationHandler: CustomerIOPushNotificationHandler
+    private lateinit var pushNotificationPayload: CustomerIOParsedPushPayload
+
+    @Before
+    override fun setup() {
+        super.setup()
+
+        val extras = Bundle.EMPTY
+        pushNotificationHandler = CustomerIOPushNotificationHandler(mock(), RemoteMessage(extras))
+        pushNotificationPayload = CustomerIOParsedPushPayload(
+            extras = extras,
+            deepLink = String.random,
+            cioDeliveryId = String.random,
+            cioDeliveryToken = String.random,
+            title = String.random,
+            body = String.random
+        )
+    }
+
+    @Test
+    fun createIntentForNotificationClick_validPayload_shouldStartDeepLinkedActivity() {
```

Comment:
Would it add value if there was another test function added where `given not valid payload, should not start Activity`?

Reply:
No actually. The activity should be started in all cases because this is being triggered when notification is clicked. So even if the payload is broken, Android OS still need something to start when user taps the notification. Whether we open anything next or not will be decided by `NotificationClickReceiverActivity`.

But I see the test name is confusing, I have updated it.